btc-types/src/zcash_header.rs

```diff
@@ -0,0 +1,206 @@
+use borsh::{BorshDeserialize, BorshSerialize};
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    hash::{double_sha256, H256},
+    u256::U256,
+    utils::serd_u32_hex,
+};
+
+type Target = U256;
+type Work = U256;
+pub type Error = crate::utils::DecodeHeaderError;
+
+#[derive(BorshDeserialize, BorshSerialize, Serialize, Deserialize, Clone, Debug, PartialEq, Eq)]
+#[serde(rename_all = "camelCase")]
+pub struct Header {
+    /// Block version, now repurposed for soft fork signalling.
+    pub version: i32,
+    /// Reference to the previous block in the chain.
+    pub prev_block_hash: H256,
+    /// The root hash of the merkle tree of transactions in the block.
+    pub merkle_root: H256,
+    /// The root hash of the block commitments tree.
+    pub block_commitments: H256,
+    /// The timestamp of the block, as claimed by the miner.
+    pub time: u32,
+    /// The target value below which the blockhash must lie.
+    #[serde(with = "serd_u32_hex")]
+    pub bits: u32,
+    // The block's nonce (Zcash: 32 bytes)
+    pub nonce: H256,
+    /// The block solution (Zcash: Equihash solution)
+    #[serde(with = "hex::serde")]
+    pub solution: Vec<u8>,
```

Comment:
The Zcash headers consume a lot of storage, I think we can avoid storing the solution field

Reply:
Fixed in https://github.com/Near-One/btc-light-client-contract/pull/97/commits/ea30eab478ad328813961fd16ea578069f4cc88e